src/log_test.ts

```diff
@@ -1,8 +1,177 @@
+import { Writable } from 'node:stream';
 
 describe('Log', () => {
+    describe('Constructor', () => {
+        it('should work', () => {
+            const config = new KubeConfig();
+            config.addCluster({
+                name: 'foo',
+                server: 'https://example.com',
+                caData: 'certificate-authority-data',
+                skipTLSVerify: false,
+            });
+            const log = new Log(config);
+            strictEqual(log.config, config);
+        });
+    });
+    describe('log', () => {
+        const config = new KubeConfig();
+        config.addCluster({
+            name: 'foo',
+            server: 'https://example.com',
+            caData: 'certificate-authority-data',
+            skipTLSVerify: false,
+        });
+        config.addContext({
+            name: 'foo',
+            cluster: 'foo',
+            user: 'foo',
+        });
+        config.setCurrentContext('foo');
+        const log = new Log(config);
+
+        afterEach(() => {
+            nock.cleanAll();
+        });
+
+        it('should make a request with correct parameters', async () => {
+            const namespace = 'default';
+            const podName = 'mypod';
+            const containerName = 'mycontainer';
+            const stream = new Writable({
+                write(chunk, encoding, callback) {
+                    callback();
+                },
+            });
+            const options: LogOptions = {
+                follow: true,
+                limitBytes: 100,
+                pretty: true,
+                previous: true,
+                sinceSeconds: 1,
+                tailLines: 1,
+                timestamps: true,
+            };
+
+            nock('https://example.com')
+                .get('/api/v1/namespaces/default/pods/mypod/log')
+                .query({
+                    container: 'mycontainer',
+                    follow: 'true',
+                    limitBytes: '100',
+                    pretty: 'true',
+                    previous: 'true',
+                    sinceSeconds: '1',
+                    tailLines: '1',
+                    timestamps: 'true',
+                })
+                .reply(200, 'log data');
+
+            const controller = await log.log(namespace, podName, containerName, stream, options);
+            strictEqual(controller instanceof AbortController, true);
+        });
+
+        it('should throw an error if no active cluster', async () => {
+            const configWithoutCluster = new KubeConfig();
+            const logWithoutCluster = new Log(configWithoutCluster);
+            const namespace = 'default';
+            const podName = 'mypod';
+            const containerName = 'mycontainer';
+            const stream = new Writable({
+                write(chunk, encoding, callback) {
+                    callback();
+                },
+            });
+
+            rejects(async () => {
```

Comment:
`rejects()` returns a Promise. We either need to `await` it or return it (I believe mocha supports that). Otherwise, if this assertion fails, it will lead to an unhandled rejection, which mocha ignores (I recently had to debug this exact case when a team was migrating from mocha to node:test).

Reply:
done.